pkg/apis/core/v1alpha1/defaults.go

```diff
@@ -176,12 +173,7 @@ func SetDefaults_Shoot(obj *Shoot) {
 		obj.Spec.Addons.KubernetesDashboard = &KubernetesDashboard{}
 	}
 	if obj.Spec.Addons.KubernetesDashboard.AuthenticationMode == nil {
-		var defaultAuthMode string
-		if *obj.Spec.Kubernetes.KubeAPIServer.EnableBasicAuthentication {
-			defaultAuthMode = KubernetesDashboardAuthModeBasic
```

Comment:
You could drop this constant now, right?

Reply:
Yes, I think we can remove it. Previously I had concerns whether it can cause issues and fail validation if the Shoot spec has the field with value `basic`.
But in https://github.com/gardener/gardener/blob/8d53342ddcf0112209ce85662d7ba6bbcf2890aa/vendor/k8s.io/apiserver/pkg/registry/rest/update.go#L129-L157 and https://github.com/gardener/gardener/blob/8d53342ddcf0112209ce85662d7ba6bbcf2890aa/vendor/k8s.io/apiserver/pkg/registry/rest/create.go#L120-L124 the PrepareForUpdate and PrepareForCreate funcs are called before the Validate funcs. We set the field always to token in PrepareForUpdate/PrepareForCreate, hence it should not cause troubles with the validation.